tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -19,111 +19,232 @@ func CreateNodes(client *rancher.Client, numOfInstances int) ([]*nodes.Node, err
-		ImageId:      aws.String(ec2Client.Config.AWSAMI),
-		InstanceType: aws.String(ec2Client.Config.InstanceType),
-		MinCount:     aws.Int64(int64(numOfInstances)),
-		MaxCount:     aws.Int64(int64(numOfInstances)),
-		KeyName:      aws.String(sshName),
-		BlockDeviceMappings: []*ec2.BlockDeviceMapping{
-			{
-				DeviceName: aws.String("/dev/sda1"),
-				Ebs: &ec2.EbsBlockDevice{
-					VolumeSize: aws.Int64(int64(ec2Client.Config.VolumeSize)),
+	var ec2Nodes []*nodes.Node
+	for _, config := range ec2Client.ClientConfig.AWSEC2Config {
+		if !config.IsWindows {
+			sshName := getSSHKeyName(config.AWSSSHKeyName)
+			runInstancesInput := &ec2.RunInstancesInput{
+				ImageId:      aws.String(config.AWSAMI),
+				InstanceType: aws.String(config.InstanceType),
+				MinCount:     aws.Int64(int64(numOfInstances)),
+				MaxCount:     aws.Int64(int64(numOfInstances)),
+				KeyName:      aws.String(sshName),
+				BlockDeviceMappings: []*ec2.BlockDeviceMapping{
+					{
+						DeviceName: aws.String("/dev/sda1"),
+						Ebs: &ec2.EbsBlockDevice{
+							VolumeSize: aws.Int64(int64(config.VolumeSize)),
+						},
+					},
 				},
-			},
-		},
-		IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
-			Name: aws.String(ec2Client.Config.AWSIAMProfile),
-		},
-		Placement: &ec2.Placement{
-			AvailabilityZone: aws.String(ec2Client.Config.AWSRegionAZ),
-		},
-		NetworkInterfaces: []*ec2.InstanceNetworkInterfaceSpecification{
-			{
-				DeviceIndex:              aws.Int64(0),
-				AssociatePublicIpAddress: aws.Bool(true),
-				Groups:                   aws.StringSlice(ec2Client.Config.AWSSecurityGroups),
-			},
-		},
-		TagSpecifications: []*ec2.TagSpecification{
-			{
-				ResourceType: aws.String("instance"),
-				Tags: []*ec2.Tag{
+				IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
+					Name: aws.String(config.AWSIAMProfile),
+				},
+				Placement: &ec2.Placement{
+					AvailabilityZone: aws.String(config.AWSRegionAZ),
+				},
+				NetworkInterfaces: []*ec2.InstanceNetworkInterfaceSpecification{
 					{
-						Key:   aws.String("Name"),
-						Value: aws.String(nodeBaseName),
+						DeviceIndex:              aws.Int64(0),
+						AssociatePublicIpAddress: aws.Bool(true),
+						Groups:                   aws.StringSlice(config.AWSSecurityGroups),
 					},
+				},
+				TagSpecifications: []*ec2.TagSpecification{
 					{
-						Key:   aws.String("CICD"),
-						Value: aws.String(ec2Client.Config.AWSCICDInstanceTag),
+						ResourceType: aws.String("instance"),
+						Tags: []*ec2.Tag{
+							{
+								Key:   aws.String("Name"),
+								Value: aws.String(nodeBaseName),
+							},
+							{
+								Key:   aws.String("CICD"),
+								Value: aws.String(config.AWSCICDInstanceTag),
+							},
+						},
 					},
 				},
-			},
-		},
-	}
+			}
 
-	reservation, err := ec2Client.SVC.RunInstances(runInstancesInput)
-	if err != nil {
-		return nil, err
-	}
+			reservation, err := ec2Client.SVC.RunInstances(runInstancesInput)
```

Comment:
Some of the spacing in this file is strange. Either remove the tabs that shouldn't be there, or run a `go.fmt` on your files to standardize the formatting